cclib/parser/orcaparser.py

```diff
@@ -49,10 +49,120 @@ def before_parsing(self):
+        # ================================================================================
+        if "INPUT FILE" == line.strip():
+            self.skip_line(inputfile, '=')
+            name = next(inputfile).split()[-1]
+            methods = []
+            line = next(inputfile)
+            coords = []
+            while line[0] == '|':
+                line = line[6:]
+
+                # Keywords block
+                if line[0] == '!':
+                    methods += line[1:].split()
+
+                # Impossible to parse without knowing whether a keyword opens a new block
+                elif line[0] == '%':
+                    pass
+
+                # Geometry block
+                elif line[0] == '*':
+                    vals = line[1:].split()
+                    coord_type = vals[0].lower()
+                    if coord_type == 'xyz':
+                        def splitter(line):
+                            atom, x, y, z = line.split()[:4]
+                            return [atom, float(x), float(y), float(z)]
+                    elif coord_type in ['int', 'internal']:
+                        def splitter(line):
+                            atom, a1, a2, a3, bond, angle, dihedral = line.split()[:7]
+                            return [atom, int(a1), int(a2), int(a3), float(bond), float(angle), float(dihedral)]
+                    elif coord_type == 'gzmt':
+                        def splitter(line):
+                            vals = line.split()[:7]
+                            if len(vals) == 7:
+                                atom, a1, bond, a2, angle, a3, dihedral = vals
+                                return [atom, int(a1), float(bond), int(a2), float(angle), int(a3), float(dihedral)]
+                            elif len(vals) == 5:
+                                return [vals[0], int(vals[1]), float(vals[2]), int(vals[3]), float(vals[4])]
+                            elif len(vals) == 3:
+                                return [vals[0], int(vals[1]), float(vals[2])]
+                            elif len(vals) == 1:
+                                return [vals[0]]
+                            self.logger.warning('Incorrect number of atoms in input geometry.')
+                    elif 'file' in coord_type:
+                        pass
+                    else:
+                        self.logger.warning('Invalid coordinate type.')
+
+                    line = next(inputfile)[6:].strip()
+                    if 'file' not in coord_type:
+                        while line[0] != '*':
+                            # Strip basis specification that can appear after coordinates
+                            line = line.split('newGTO')[0]
+                            coords.append(splitter(line))
+                            line = next(inputfile)[6:].strip()
+
+                line = next(inputfile)
+
+            self.metadata['methods'] = methods
```

Comment:
I think that `keywords` would be a better name, since most entries there are not actually methods, though we don't currently have a way of separating the two. Do you have any ideas? My current idea is to do a search-and-replace after parsing with dictionaries of methods and keywords, maybe as another library...